### PR TITLE
MM-17394: tune /jira assign user search

### PR DESCRIPTION
**Pull request by levb**
- Use the rest of the command line for search, rather than just the
  last (4th) argument.
- Display username and email when available (Jira server, and Cloud when
  email is public
- Style/naming

Jira ticket: https://mattermost.atlassian.net/browse/MM-17394

### CODE REVIEW

File: server/issue.go [hunk 1]
```diff
@@ -647,19 +648,18 @@ func (p *Plugin) assignJiraIssue(mmUserId, issueKey, assignee string) (string, e
...
 	}
 
-	// Get list of assignable assignees
-	url := fmt.Sprintf("rest/api/2/user/assignable/search?issueKey=%s&username=%s", issueKey, assignee)
-	req, err := jiraClient.NewRequest("GET", url, nil)
-	if err != nil {
-		return "", err
+	// Get list of assignable users
+	var jiraUsers []jira.User
+	status, err := JiraGet(jiraClient, "user/assignable/search",
```
Comment (mickmister) [inline server/issue.go]: :+1: for `JiraGet`

File: server/issue.go [hunk 2]
```diff
@@ -647,19 +648,18 @@ func (p *Plugin) assignJiraIssue(mmUserId, issueKey, assignee string) (string, e
...
+		map[string]string{
+			"issueKey":   issueKey,
+			"username":   userSearch,
+			"maxResults": "10",
```
Comment (mickmister) [inline server/issue.go]: :+1: for `maxResults: 10`

File: server/issue.go [hunk 3]
```diff
@@ -819,3 +819,31 @@ func attachFileToIssue(api plugin.API, jiraClient *jira.Client, issueKey, fileId
 	attachment := (*attachments)[0]
 	return fileinfo.Name, attachment.Filename, nil
 }
+
+func JiraGet(jiraClient *jira.Client, api string, params map[string]string, dest interface{}) (int, error) {
+	return jiraGet(jiraClient, 2, api, params, dest)
+}
+
+func JiraGet2(jiraClient *jira.Client, api string, params map[string]string, dest interface{}) (int, error) {
```
Comment (mickmister) [inline server/issue.go]: Unused function?

File: server/issue.go (hunk 1 above)
Comment (mickmister) [reply server/issue.go]: Note, my linter warns me of dead code. Should we hook that sort of linting into CI?

File: server/issue.go (hunk 3 above)
Comment (mickmister) [reply server/issue.go]: Note, my linter warns me of dead code. Should we hook that sort of linting into CI?

File: server/manifest.go [hunk 4]
```diff
@@ -7,5 +7,5 @@ var manifest = struct {
 }{
 	Id:      "jira",
 	Version: "2.1.0",
-	Hash:    "06a79da",
+	Hash:    "4b539b4",
```
Comment (mickmister) [inline server/manifest.go]: :+1:

**Comment by levb** [comment]
Removed PM review - there is no UI change

**Pull request merged**